### PR TITLE
Refactor integratord `test_integratord_read_json_file_deleted`

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|-------------|
| https://github.com/wazuh/wazuh-qa/issues/3298 |

## Description

This PR aims to add some changes to the development made by @mauromalara in this [PR](https://github.com/wazuh/wazuh-qa/pull/3362).

### Added

- `tests/integration/test_integratord/test_alerts_reading.py`: All modules were unified in this one and several changes were applied in the tests functions and documentation.

### Updated

- `deps/wazuh_testing/modules/analysisd/__init__.py`: Callback added to detect its initialization.
- `deps/wazuh_testing/modules/integratord/__init__.py`: Callbacks modified to fix the tests.
- `deps/wazuh_testing/modules/integratord/event_monitor.py`: Callback functions modified and documentation improved.
- `deps/wazuh_testing/wazuh_testing/tools/__init__.py`: Integratord socket added.
- `tests/integration/conftest.py`: Fixture improved and Slack webhook parameter added. Daemons handler modified and wrapper with 'function' scope added.
- `tests/integration/test_integratord/conftest.py`: Minor improvements.
- `tests/integration/test_integratord/data/configuration/configuration_alerts_reading.yaml`: Configuration template modified to change the integration from VTotal to Slack.
- `tests/integration/test_integratord/data/test_cases/cases_integratord_change_inode_alert.yaml`: Virustotal integration replaced.
- `tests/integration/test_integratord/data/test_cases/cases_integratord_read_invalid_json_alerts.yaml`: Virustotal integration replaced.
- `tests/integration/test_integratord/data/test_cases/cases_integratord_read_valid_json_alerts.yaml`: Virustotal integration replaced.

### Deleted

- `tests/integration/test_integratord/data/test_cases/cases_integratord_read_json_file_deleted.yaml`: Test case deleted because it was duplicated by `test_change_inode_alert`
- `tests/integration/test_integratord/test_integratord_read_json_file_deleted.py`: Test deleted because it was duplicated by `test_change_inode_alert`
- `tests/integration/test_integratord/test_integratord_change_inode_alert.py`: All modules were unified in `test_alerts_reading.py`
- `tests/integration/test_integratord/test_integratord_read_json_alerts.py`: All modules were unified in `test_alerts_reading.py`
- `tests/integration/test_integratord/test_integratord_read_json_alerts.py`: All modules were unified in `test_alerts_reading.py`

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @mauromalara (Developer)  | test_integratord/ | [🟢](https://devel.ci.wazuh.info/job/Test_integration_VR/30/) [🟢](https://devel.ci.wazuh.info/job/Test_integration_VR/31/console) [🟢](https://devel.ci.wazuh.info/job/Test_integration_VR/32/console) | [🟢](https://github.com/wazuh/wazuh-qa/files/10675584/report-1.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/10675585/report-2.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/10675586/report-3.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/10675587/report-4.zip) [🟢](https://github.com/wazuh/wazuh-qa/files/10675588/report-5.zip) | Ubuntu | 7b78001 | - |
| @Rebits  (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_integratord/conftest.py [hunk 1]
```diff
@@ -8,15 +8,22 @@
...
 
 
 @pytest.fixture(scope='function')
 def wait_for_start_module(request):
     # Wait for integratord thread to start
     file_monitor = FileMonitor(LOG_FILE_PATH)
-    check_integratord_event(file_monitor=file_monitor, timeout=20,
-                            callback=generate_monitoring_callback(integrator.CB_INTEGRATORD_THREAD_READY),
-                            error_message=integrator.ERR_MSG_VIRUST_TOTAL_ENABLED_NOT_FOUND)
+    check_integratord_event(file_monitor=file_monitor, timeout=T_20,
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply tests/integration/test_integratord/conftest.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3540/commits/cef625211244aafe84c6e5dc4cdb6410786ddb51)

File: deps/wazuh_testing/wazuh_testing/modules/integratord/__init__.py [hunk 2]
```diff
@@ -3,29 +3,31 @@
...
-CB_INTEGRATORD_SENDING_ALERT = r'(.*wazuh-integratord.*DEBUG: sending new alert).*'
-CB_PROCESSING_ALERT = r'.*(wazuh-integratord.*Processing alert.*)'
-CB_INTEGRATORD_THREAD_READY = r'.*(wazuh-integratord.*DEBUG: Local requests thread ready).*'
-CB_VIRUSTOTAL_ALERT = r'.*(wazuh-integratord.*alert_id.*\"integration\": \"virustotal\").*'
-CB_VIRUSTOTAL_ALERT_JSON = r'.*(VirusTotal: Alert.*\"integration\":\"virustotal\").*'
-CB_INVALID_JSON_ALERT_READ = r'.*(wazuh-integratord.*WARNING: Invalid JSON alert read).*'
-CB_OVERLONG_JSON_ALERT_READ = r'.*(wazuh-integratord.*WARNING: Overlong JSON alert read).*'
-CB_ALERTS_FILE_INODE_CHANGED = r'.*(wazuh-integratord.*DEBUG: jqueue_next.*Alert file inode changed).*'
-CB_CANNOT_RETRIEVE_JSON_FILE = r'.*(wazuh-integratord.*WARNING.*Could not retrieve information of file).*'\
+CB_SLACK_ENABLED = fr".*{INTEGRATORD_PREFIX}.*Enabling integration for: 'slack'.*"
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply deps/wazuh_testing/wazuh_testing/modules/integratord/__init__.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3540/commits/cef625211244aafe84c6e5dc4cdb6410786ddb51)

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py [hunk 3]
```diff
@@ -873,6 +873,15 @@ def new_callback(line):
     return new_callback
 
 
+def callback_generator(regex):
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply deps/wazuh_testing/wazuh_testing/tools/monitoring.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3540/commits/cef625211244aafe84c6e5dc4cdb6410786ddb51)

**Comment by mauromalara** [comment]
Before merging these changes, it is necessary to merge [this PR](https://github.com/wazuh/wazuh-jenkins/pull/4023) in the "wazuh-jenkins" repository.
> More info: https://github.com/wazuh/wazuh-jenkins/issues/3838#issuecomment-1263990324

Also, it is necessary to merge [the backport to 4.5](https://github.com/wazuh/wazuh-jenkins/pull/3860) and create a backport of [this PR](https://github.com/wazuh/wazuh-jenkins/pull/4023) to `4.5`

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -293,12 +321,12 @@ def pytest_addoption(parser):
         help="run tests using a specific WPK package path"
     )
     parser.addoption(
-        "--integration-api-key",
+        "--slack-webhook-url",
```
[the remark this answers is not in the document]
Comment (mauromalara) [reply tests/integration/conftest.py:324]: PR: https://github.com/wazuh/wazuh-jenkins/pull/4023

**Comment by mauromalara** [comment]
### Wazuh Jenkins dependency

This PR depends on some changes to be merged in the `wazuh-jenkins` repo in the following order:

- [ ] https://github.com/wazuh/wazuh-jenkins/pull/4689
- [ ] wazuh/wazuh-jenkins#4690
- [ ] wazuh/wazuh-jenkins#4691
- [ ] wazuh/wazuh-jenkins#4692

**Comment by mauromalara** [comment]
### IT executions

|R1|R2|R3
|--|--|--
| [🟢](https://ci.wazuh.info/job/Test_integration/36390/) | [🟢](https://ci.wazuh.info/job/Test_integration/36391/) | [🟢](https://ci.wazuh.info/job/Test_integration/36392/)

File: tests/integration/conftest.py [hunk 5]
```diff
@@ -123,11 +123,39 @@ def restart_wazuh_daemon_function(daemon=None):
...
+    # Stop all daemons by default (daemons = None)
+    if len(daemons) == 0:
+        logger.debug(f"Stopping all daemons")
+        control_service('stop')
+    else:
+        # Stop a list daemons in order (as Wazuh does)
+        daemons.reverse()
+        for daemon in daemons:
+            logger.debug(f"Stopping {daemon}")
+            control_service('stop', daemon=daemon)
```
[the remark this answers is not in the document]
Comment (mauromalara) [reply tests/integration/conftest.py:158]: IT Executions: https://github.com/wazuh/wazuh-qa/pull/3540#issuecomment-1422299461

**Comment by mauromalara** [comment]
## Dependency

- Merge #3915

**Review by juliamagan — CHANGES_REQUESTED** [review]
Please update branches and launch the tests to check that everything works as expected.

**Comment by mauromalara** [comment]
## Last results

| R1 | R2 | R3 |
|--------|--------|--------|
| [🟢](https://github.com/wazuh/wazuh-qa/files/11447407/Archive.2.zip) | [🟢](https://github.com/wazuh/wazuh-qa/files/11447408/Archive.3.zip) | [🟢](https://github.com/wazuh/wazuh-qa/files/11447406/Archive.zip) |

### Note ⚠️ 

This PR is blocked by https://github.com/wazuh/wazuh-qa/issues/4126 because it depends on that fix to get merged.

**Comment by mauromalara** [comment]
## After merge

https://github.com/wazuh/wazuh-qa/issues/4165

**Review by Rebits — APPROVED** [review]
LGTM

**Pull request merged**